stream-chat-android-client/src/test/java/io/getstream/chat/android/client/experimental/socket/lifecycle/LifecyclePublisherTest.kt

```diff
@@ -0,0 +1,219 @@
+
+        combinedLifecycleEvents.size `should be equal to` 2
+        combinedLifecycleEvents[0] `should be` Event.Lifecycle.Started
+        combinedLifecycleEvents[1] `should be instance of` Event.Lifecycle.Stopped.WithReason::class
+    }
+
+    @Test
+    fun `given all stopped when not all publisher start should not emit lifecycle start`(): Unit = runTest {
+        // Given all lifecycle stopped.
+        lifecyclePublisher1.sendLifecycleStoppedEvent()
+        lifecyclePublisher2.sendLifecycleStoppedEvent()
+        lifecyclePublisher3.sendLifecycleStoppedEvent()
+
+        combinedLifecycleEvents.size `should be equal to` 1
+        combinedLifecycleEvents[0] `should be instance of` Event.Lifecycle.Stopped::class
+
+        // Sending lifecycle start event again to publisher2.
+        lifecyclePublisher2.sendLifecycleStartedEvent()
+
+        combinedLifecycleEvents.size `should be equal to` 1
+        combinedLifecycleEvents[0] `should be instance of` Event.Lifecycle.Stopped::class
+
+        // Sending lifecycle start event again to publisher1.
+        lifecyclePublisher1.sendLifecycleStartedEvent()
+
+        combinedLifecycleEvents.size `should be equal to` 1
+        combinedLifecycleEvents[0] `should be instance of` Event.Lifecycle.Stopped::class
+    }
+
+    @Test
+    fun `given lifecycle stopped when all publisher start should emit lifecycle start`(): Unit = runTest {
+
+        // Given all lifecycle stopped.
+        lifecyclePublisher1.sendLifecycleStoppedEvent()
+        lifecyclePublisher2.sendLifecycleStoppedEvent()
+        lifecyclePublisher3.sendLifecycleStoppedEvent()
+
+        combinedLifecycleEvents.size `should be equal to` 1
+        combinedLifecycleEvents[0] `should be instance of` Event.Lifecycle.Stopped::class
+
+        // Sending lifecycle start event again to publisher2.
+        lifecyclePublisher2.sendLifecycleStartedEvent()
+
+        combinedLifecycleEvents.size `should be equal to` 1
+        combinedLifecycleEvents[0] `should be instance of` Event.Lifecycle.Stopped::class
+
+        // Sending lifecycle start event again to publisher1.
+        lifecyclePublisher1.sendLifecycleStartedEvent()
+
+        combinedLifecycleEvents.size `should be equal to` 1
+        combinedLifecycleEvents[0] `should be instance of` Event.Lifecycle.Stopped::class
+
+        // Sending lifecycle start event again to publisher3.
+        lifecyclePublisher3.sendLifecycleStartedEvent()
+
+        combinedLifecycleEvents.size `should be equal to` 2
+        combinedLifecycleEvents[0] `should be instance of` Event.Lifecycle.Stopped::class
+        combinedLifecycleEvents[1] `should be` Event.Lifecycle.Started
+    }
+
+    @Test
+    fun `given any stopAndAbort should emit first stopAndAbort event`(): Unit = runTest {
+        lifecyclePublisher3.sendLifecycleStoppedEvent(Event.Lifecycle.Stopped.WithReason(cause = DisconnectCause.ConnectionReleased))
+        delay(1000)
```

Comment:
`delay()` functions are skipped when running whitin a `runTest{}` lambda, so they shouldn't be needed here